### PR TITLE
[NavigationMenuTrigger] Fix positioner height when opening the list using the keyboard

### PR DESCRIPTION
**Pull request by juliomerisio**
**Before**

https://github.com/user-attachments/assets/93f82d76-c980-43c9-9c3b-b15307b615d8

**After**

https://github.com/user-attachments/assets/2abc0f3c-cfd2-4db2-9002-059b0fc0d3d0

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/base-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

**Review by atomiks — APPROVED** [review]
The fix works correctly 👍 

I'll let @michaldudak decide on the infra addition - we usually write our tests with Vitest browser mode with RTL if we need to calculate real sizes for in the browser at the moment (using `skipIf(isJSDOM)`), which is why the e2e is largely empty

File: test/e2e/index.test.ts [hunk 1]
```diff
@@ -129,4 +132,48 @@ describe('e2e', () => {
       await expect(screen.getByTestId('three')).toHaveFocus();
     });
   });
+
+  describe('<NavigationMenu />', () => {
```
Comment (michaldudak) [inline test/e2e/index.test.ts]: This test can be added to our unit test suite. We run it in the browser using Vitest's browser mode (using Playwright under the hood).

File: test/e2e/fixtures/NavigationMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      />
+      <path
+        d="M10.3333 3.34539L5.47654 7.71648C4.55842 8.54279 3.36693 9 2.13172 9H0V8H2.13172C3.11989 8 4.07308 7.63423 4.80758 6.97318L9.66437 2.60207C10.0447 2.25979 10.622 2.2598 11.0023 2.60207L15.8591 6.97318C16.5936 7.63423 17.5468 8 18.5349 8H20V9H18.5349C17.2998 9 16.1083 8.54278 15.1901 7.71648L10.3333 3.34539Z"
+        className="dark:fill-gray-300"
+      />
+    </svg>
+  );
+}
+
+const triggerClassName =
```
Comment (michaldudak) [inline test/e2e/fixtures/NavigationMenu.tsx]: We don't use Tailwind CSS in our tests. Also, we don't need all the bells and whistles in test code, only the minimum to verify the functionality.

Comment (juliomerisio) [reply test/e2e/fixtures/NavigationMenu.tsx]: Good to know! code removed.

File: test/e2e/index.test.ts (hunk 1 above)
Comment (juliomerisio) [reply test/e2e/index.test.ts]: Nice! I just replaced the Playwright test with the unit test.

File: packages/react/src/navigation-menu/trigger/NavigationMenuTrigger.test.tsx [hunk 3]
```diff
@@ -17,4 +19,75 @@ describe('<NavigationMenu.Trigger />', () => {
...
   }));
+
+  it('handles focus and positioner height', async () => {
+    await render(
+      <NavigationMenu.Root>
+        <NavigationMenu.List>
+          <NavigationMenu.Item>
+            <NavigationMenu.Trigger>Overview</NavigationMenu.Trigger>
+            <NavigationMenu.Content>
+              <NavigationMenu.Link href="#">Quick Start Install and</NavigationMenu.Link>
```
Comment (michaldudak) [inline packages/react/src/navigation-menu/trigger/NavigationMenuTrigger.test.tsx]:
```suggestion
              <NavigationMenu.Link href="#">Quick Start</NavigationMenu.Link>
```
